### DISCUSSION TITLE
NextJS Memory Leak

### DISCUSSION THREAD

**Discussion by antoniofpereir** [vercel/next.js#66986] [Help]
### Summary

Running nextjs (v14.2.3) in a docker instance causes memory leaks (heap dump examples below)
![image](https://github.com/vercel/next.js/assets/26146029/28c1b161-06cb-4979-bdcb-4b821d29585d)

We already optimized the code as much as we could but taking heap snapshots from a local docker instance running a stress test shows an accumulation of the "(string)" mainly in "__ next_chunk_load__"(examples below) and upon opening them it's just values that we pass as props to multiple components (server side rendered)
![image](https://github.com/vercel/next.js/assets/26146029/4b2ca120-02e9-46fa-bda0-d1bcddd7fefc)
![image](https://github.com/vercel/next.js/assets/26146029/71be0ecd-c1e4-4860-8339-c78a6e444113)

### Additional information

```js
Operating System: node 18 alpine (18.20.3)
NextJS Version: 14.2.3
```

**Comment by chipcop106** [comment]
I have the same issues. Memory leak with NextJS 14.2.3 and node 18 alpine (18.20.3). Any one help?
<img width="1089" alt="image" src="https://github.com/vercel/next.js/assets/17249557/19a73f4c-dda0-41a1-82f0-cc465ebea55a">

**Comment by antoniofpereir** [comment]
I got some update, it turns out it was not a problem on NextJS side. We were performing a fetch request without using .json, .blob or .text, which prevents the request from closing and ends up lingering in memory, worth a check

**Reply by AdaoNatalino** [reply]
I'm having the same issue, probably not caused by not parsing, since I'm using https://openapi-ts.pages.dev/openapi-fetch/#basic-usage, so assume the response is parsed already and ready to use, as per example.`
🤔

**Reply by VIGDXX** [reply]
We were performing a fetch request without using .json, .blob or .text, which prevents the request from closing and ends up lingering in memory, worth a check. --- I'm having the same issue, but i can't understand the solution,can you show more details please?

**Reply by YutaMoriJP** [reply]
@VIGDXX I think what they're saying is that you must read the body: 

This would leak in Node. So you have to read the body `json/text`, etc. If you just need the `headers` of the response, then you should just fetch the headers.
```js
const res = await fetch();
if(!res.ok) {
throw new Error()
}
```

**Reply by LokeshParthiban** [reply]
I use Next v14.2.4 with node 18.x. I face similar memory leak issues in my server where nextjs is running inside docker. We do read all the fetch responses with .json() or .text(). But i still see the leak.

**Reply by abemelek** [reply]
> I use Next v14.2.4 with node 18.x. I face similar memory leak issues in my server where nextjs is running inside docker. We do read all the fetch responses with .json() or .text(). But i still see the leak.

I had the same issue. What fixed it was bumping my docker base image to node:20-alpine3.21 from node:20-alpine3.18

**Reply by twoyoung04** [reply]
> > I use Next v14.2.4 with node 18.x. I face similar memory leak issues in my server where nextjs is running inside docker. We do read all the fetch responses with .json() or .text(). But i still see the leak.
> 
> I had the same issue. What fixed it was bumping my docker base image to node:20-alpine3.21 from node:20-alpine3.18

it works for me too
